core/src/test/scala/shapeless/serialization.scala

```diff
@@ -937,11 +943,11 @@ class SerializationTests {
   def testLazy {
     assertSerializable(Lazy(23))
 
-    assertSerializable(implicitly[Lazy[Generic[Wibble]]])
-    assertSerializable(implicitly[Lazy[Generic1[Box, TC1]]])
+    assertSerializableBeforeAfter(implicitly[Lazy[Generic[Wibble]]])(_.value)
+    assertSerializableBeforeAfter(implicitly[Lazy[Generic1[Box, TC1]]])(_.value)
 
```

Comment:
Ouch. Yes, I guess that makes sense. I still think that we shouldn't be capturing that outer reference though.